src/item.cpp

```diff
@@ -3325,20 +3364,19 @@ void item::on_wield( player &p, int mv )
 
 void item::handle_pickup_ownership( Character &c )
 {
-    faction *yours = c.get_faction();
-    if( !yours ) {
+    if( is_owned_by( c ) ) {
         return;
     }
     // Add ownership to item if unowned
     if( !has_owner() ) {
-        set_owner( yours->id );
+        set_owner( c );
     } else {
-        if( get_owner() != yours->id && &c == &g->u ) {
+        if( !is_owned_by( c ) && &c == &g->u ) {
             std::vector<npc *> witnesses;
             for( npc &elem : g->all_npcs() ) {
                 if( rl_dist( elem.pos(), g->u.pos() ) < MAX_VIEW_DISTANCE && elem.get_faction() &&
                     elem.get_faction()->id != faction_id( "no_faction" ) &&
-                    elem.get_faction()->id == get_owner() && elem.sees( g->u.pos() ) ) {
+                    is_owned_by( elem ) && elem.sees( g->u.pos() ) ) {
```

Comment:
again, collapse `elem.get_faction()->id != faction_id( "no_faction" ) && is_owned_by( elem )` into `is_owned_by()`, possibly with an optional argument to differentiate between items that have no owner and items that are owned by someone else.